src/icland/__init__.py

```diff
@@ -147,7 +147,16 @@ def init(key: jax.Array, params: ICLandParams) -> ICLandState:
             [body_id, geom_id, dof_address]
         )
 
-    return ICLandState(mjx_model, mjx_data, agent_components)
+    pipeline_state = PipelineState(mjx_model, mjx_data, agent_components)
+
+    return ICLandState(
```

Comment:
Is it worth having ICLandState inherit from Brax State? (might not be possible if we're already inheriting from pytree) Or alternatively, just returning a `State` altogether?

I understand though that this is technically fine due to Python duck typing, but we might want to be more specific.